utils/rsaencryption/rsa_encryption.go

```diff
@@ -59,18 +62,48 @@ func DecodeKey(sk *rsa.PrivateKey, hash []byte) ([]byte, error) {
 // ConvertPemToPrivateKey return rsa private key from secret key
 func ConvertPemToPrivateKey(skPem string) (*rsa.PrivateKey, error) {
 	block, _ := pem.Decode([]byte(skPem))
-	// TODO: resolve deprecation https://github.com/golang/go/issues/8860
-	enc := x509.IsEncryptedPEMBlock(block) //nolint
+	if block == nil {
+		return nil, errors.New("Failed to decode PEM block")
+	}
 	b := block.Bytes
-	if enc {
-		var err error
-		// TODO: resolve deprecation https://github.com/golang/go/issues/8860
-		b, err = x509.DecryptPEMBlock(block, nil) //nolint
-		if err != nil {
-			return nil, errors.Wrap(err, "Failed to decrypt private key")
-		}
+	return parsePrivateKey(b)
+}
+
+// ConvertEncryptedPemToPrivateKey return rsa private key from secret key
+func ConvertEncryptedPemToPrivateKey(pemData []byte, password string) (*rsa.PrivateKey, error) {
+	if strings.TrimSpace(password) == "" {
+		return nil, errors.New("Password required for encrypted PEM block")
+	}
+
+	// Unmarshal the JSON-encoded data
+	var data map[string]interface{}
+	if err := json.Unmarshal(pemData, &data); err != nil {
+		return nil, errors.Wrap(err, "Failed to parse JSON data")
```

Comment:
`"github.com/pkg/errors"` is discontinued, let's use Go 1.13's error handling for new errors.

```suggestion
		return nil, fmt.Errorf("parse JSON data: %w", err)
```